python/lsst/afw/detection/peak.cc

```diff
@@ -35,40 +37,75 @@ namespace lsst {
-
-    clsPeakRecord.def("getTable", &PeakRecord::getTable);
-    clsPeakRecord.def("getId", &PeakRecord::getId);
-    clsPeakRecord.def("setId", &PeakRecord::setId);
-    clsPeakRecord.def("getIx", &PeakRecord::getIx);
-    clsPeakRecord.def("getIy", &PeakRecord::getIy);
-    clsPeakRecord.def("setIx", &PeakRecord::setIx);
-    clsPeakRecord.def("setIy", &PeakRecord::setIy);
-    clsPeakRecord.def("getI", &PeakRecord::getI);
-    clsPeakRecord.def("getCentroid", (afw::geom::Point2I (PeakRecord::*)(bool) const) &PeakRecord::getCentroid);
-    clsPeakRecord.def("getCentroid", (afw::geom::Point2D (PeakRecord::*)() const) &PeakRecord::getCentroid);
-    clsPeakRecord.def("getFx", &PeakRecord::getFx);
-    clsPeakRecord.def("getFy", &PeakRecord::getFy);
-    clsPeakRecord.def("setFx", &PeakRecord::setFx);
-    clsPeakRecord.def("setFy", &PeakRecord::setFy);
-    clsPeakRecord.def("getF", &PeakRecord::getF);
-    clsPeakRecord.def("getPeakValue", &PeakRecord::getPeakValue);
-    clsPeakRecord.def("setPeakValue", &PeakRecord::setPeakValue);
-
-    py::class_<table::CatalogT<PeakRecord>, std::shared_ptr<table::CatalogT<PeakRecord>>> clsPeakRecordCatalog(mod, "PeakRecordCatalog");
-    declareCatalog(clsPeakRecordCatalog);
-    /* Module level */
+using PyPeakRecord = py::class_<PeakRecord, std::shared_ptr<PeakRecord>, table::BaseRecord>;
+using PyPeakTable = py::class_<PeakTable, std::shared_ptr<PeakTable>, table::BaseTable>;
+using PyPeakCatalog = py::class_<table::CatalogT<PeakRecord>,
+                                 std::shared_ptr<table::CatalogT<PeakRecord>>>;
+
+/**
+Declare constructors and member and static functions for a pybind11 PeakRecord
+*/
+void declarePeakRecord(PyPeakRecord & cls) {
+    table::pybind11::addCastFrom<table::BaseRecord, PeakRecord>(cls);
+
+    cls.def("getTable", &PeakRecord::getTable);
+    cls.def("getId", &PeakRecord::getId);
+    cls.def("setId", &PeakRecord::setId);
+    cls.def("getIx", &PeakRecord::getIx);
+    cls.def("getIy", &PeakRecord::getIy);
+    cls.def("setIx", &PeakRecord::setIx);
+    cls.def("setIy", &PeakRecord::setIy);
+    cls.def("getI", &PeakRecord::getI);
+    cls.def("getCentroid", (afw::geom::Point2I (PeakRecord::*)(bool) const) &PeakRecord::getCentroid);
+    cls.def("getCentroid", (afw::geom::Point2D (PeakRecord::*)() const) &PeakRecord::getCentroid);
+    cls.def("getFx", &PeakRecord::getFx);
+    cls.def("getFy", &PeakRecord::getFy);
+    cls.def("setFx", &PeakRecord::setFx);
+    cls.def("setFy", &PeakRecord::setFy);
+    cls.def("getF", &PeakRecord::getF);
+    cls.def("getPeakValue", &PeakRecord::getPeakValue);
+    cls.def("setPeakValue", &PeakRecord::setPeakValue);
+}
 
-    /* Member types and enums */
+/**
+Declare constructors and member and static functions for a pybind11 PeakTable
+*/
+void declarePeakTable(PyPeakTable & cls) {
+    table::pybind11::addCastFrom<table::BaseTable, PeakTable>(cls);
```

Comment:
Perhaps the naming is confusing. A plain `addCast` would be more readable. Or you could make it really nice and have the `To` type deduced from the `cls` (i.e. `py::class_<To, ...>`) argument.